contentcuration/contentcuration/frontend/administration/pages/Users/UserPrivilegeModal.vue

```diff
@@ -91,29 +82,47 @@
           this.$emit('input', value);
         },
       },
-      emailRules() {
-        return [
-          v => Boolean(v) || 'Field is required',
-          v => v === this.currentEmail || 'Email does not match your account email',
-        ];
+    },
+    watch: {
+      value(value) {
+        if (value) {
+          this.reset();
+        }
       },
     },
     methods: {
       close() {
         this.emailConfirm = '';
-        this.resetValidation();
+        // Clear errors manually if using generateFormMixin
+        if (this.errors) {
+          Object.keys(this.errors).forEach(key => {
+            this.errors[key] = false;
+          });
+        }
         this.dialog = false;
       },
-      resetValidation() {
-        this.$refs.form.resetValidation();
-      },
       confirm() {
-        if (this.$refs.form.validate()) {
-          return this.confirmAction();
+        if (this.onSubmit) {
+          return this.onSubmit();
         } else {
           return Promise.resolve();
         }
       },
+      // eslint-disable-next-line vue/no-unused-properties
+      onSubmit() {
+        return this.confirmAction()
+          .then(() => {
+            this.dialog = false;
+          })
+          .catch(() => {
+            this.showSnackbar({ text: this.$tr('ErrorMessage') });
```

Comment:
Just quickly from the code I don't think that snackbar was displayed before or was it? Please double-check, and if that's the case, remove this logic.